datafusion/core/src/physical_plan/coalesce_batches.rs

```diff
@@ -319,9 +318,14 @@ mod tests {
 
     #[tokio::test]
     async fn test_custom_batch_size() -> Result<()> {
-        let ctx = SessionContext::with_config(
-            SessionConfig::new().set_u64(OPT_COALESCE_TARGET_BATCH_SIZE, 1234),
-        );
+        let mut config = SessionConfig::new();
+        config
+            .config_options_mut()
```

Comment:
This will be a lot less peverse once SessionConfig is rolled into ConfigOptions #3887